APE26.rst

```diff
@@ -0,0 +1,331 @@
+remove ambiguity for users from having nearly duplicate functionality 
+with slightly different APIs, and better satisfy the principle of 
+`Separation of Concerns
+<https://en.wikipedia.org/wiki/Separation_of_concerns>`_.
+
+Detailed description
+--------------------
+The coordinate frame classes (subclasses of ``BaseCoordinateFrame``, 
+e.g., ``ICRS``, ``Galactic``, ``FK4``, etc.) are used to represent 
+astronomical reference frames. These may also contain position, 
+velocity, time, or other information about the reference frame 
+relative to another reference frame, which is used to transform 
+coordinates between the reference frames that exist in the 
+``astropy.coordinates`` ecosystem. For example, the ``AltAz`` frame 
+class, which is used when referencing sky coordinates in altitude 
+and azimuth, must contain a location on Earth and a time in order 
+to transform to and from an inertial reference frame like the ICRS. In 
+addition to reference frame information, these frame classes can 
+**also** contain coordinate data (positions, velocities, or other 
+differentials), which are stored internally using the 
+``Representation`` and ``Differential`` classes.
+
+The above reflects a design choice in the coordinate frame 
+implementation in `APE 5 
+<https://github.com/astropy/astropy-APEs/blob/main/APE5.rst>`_ 
+that, following several years of use, we have learned from and 
+think best to revise. We now view it as an insufficient 
+separation of concerns. The 
+coordinate frame class deals with two separate issues: the 
+definition of the frame and the storage of coordinate data within 
+that frame. Ideally the code would be more modularly structured, 
+where the coordinate frame class **only** defines the reference 
+frame. It would then be the purview of another class – like 
+``SkyCoord`` or the newly-proposed ``Coordinate`` – to bring these 
+concerns together and to represent data (using ``Representation`` 
+and ``Differential`` classes) in a given reference frame (using a 
+``CoordinateFrame`` class). As a demonstration of the current state 
+of duplicated functionality, these two initializations effectively 
+represent the same thing, but return different objects with similar 
+but not identical APIs:
+
+.. code-block:: python
+
+    c1 = SkyCoord(1., 2., frame="icrs", units="deg")
+    c2 = ICRS(1. * u.deg, 2. * u.deg)
+
+We suggest the situation should instead be analogous to what is the 
+case for units, which know how to transform from one to another, but 
+are not concerned with how the values are stored (that belongs to 
+``Quantity``). Translating to coordinates, units are like 
+``CoordinateFrame``, the values are the coordinate data 
+(``Representations``), and ``Quantity`` is like ``SkyCoord``.
+
+Having both the frame classes as well as ``SkyCoord`` be able to
+store and handle data has resulted in a large amount of code
+duplication. It has also required duplicated or even quadrupled 
```

Comment:
I don't think code duplication is as big of a problem as it used to be. But reducing code duplication has come at the cost of making `SkyCoord.__getattr__()` even more prevalent. It is worth mentioning that the reliance on `SkyCoord.__getattr__()` means there is at least one method that `SkyCoord` implements accidentally (https://github.com/astropy/astropy/issues/15643) and there might be more.